plasmapy/diagnostics/proton_radiography.py

```diff
@@ -576,43 +787,43 @@ def _coast_to_grid(self):
         # Coast the particles to the advanced position
         self.x = self.x + self.v * t
 
-    def _generate_null_distribution(self):
-        r"""
-        Calculate the distribution of particles on the detector plane in the absence
-        of any simulated fields.
+    def _coast_to_plane(self, center, hdir, vdir):
+        """
+        Calculates the positions where the current trajectories of each
+        particle impact a plane, described by the plane's center and
+        horizontal and vertical unit vectors.
+
+        Returns an [nparticles, 3] array of the particle positions in the plane
 
-        These positions are used to quickly compute null radiographs, which are
-        used to determine the degree of deflection.
+        Note that this function does not alter self.x itself. That is done by
+        setting self.x = the output of this function.
```

Comment:
That doesn't sound that hard - can we just do that now? Just something like 

```python

def _coast_to_plane( ... , xout = None):
    ... 
    xout  = self.x + self.v * t[:, np.newaxis]

    ...
    return xout
```
?

Reply:
Yes, but there's a trick to it: you have to do `xout[...] = rhs` or `xout[:] = rhs`. Otherwise, this will overwrite the name - the pointer to the array, if you will - rather than its contents!

Reply:
Great, that works! Is there a way to avoid doing something like this?

```python
if x is None:
            x = self.x + self.v * t[:, np.newaxis]
else:
            x[:] = self.x + self.v * t[:, np.newaxis]
```

Reply:
the only simplification possible here is probably...
```python
if x is None:
    # we didn't get any argument; let's preallocate
    x = np.empty_like(self.x) 
x[:] = self.x + self.v * t[:, np.newaxis]
```